### ISSUE TITLE
4.0.0 breaking older releases

### ISSUE THREAD

**Issue by CyberMew** [microsoft/appcenter-sdk-react-native#875]
### **Description**

Older versions are installing iOS SDK for 4.0.0, and is breaking stuff because your latest podspec for 3.1.2 is depending on the latest (which is as of 2-3 days ago, 4.0.0) which is currently incorrect as they are not compatible.

PODS:
  - appcenter-analytics (3.1.2):
    - AppCenter/Analytics
    - AppCenterReactNativeShared
    - React
  - appcenter-core (3.1.2):
    - AppCenterReactNativeShared
    - React
  - appcenter-crashes (3.1.2):
    - AppCenter/Crashes
    - AppCenterReactNativeShared
    - React
  - AppCenter/Analytics (4.0.0):
    - AppCenter/Core
  - AppCenter/Core (4.0.0)
  - AppCenter/Crashes (4.0.0):
    - AppCenter/Core
  - AppCenterReactNativeShared (4.0.0):
    - AppCenter/Core (= 4.0.0)

You should release 3.1.3 (or update 3.1.2?) which has the lines something like this in your libraries.

```
  s.dependency 'AppCenterReactNativeShared', '~> 3.0'
  s.dependency 'AppCenter/Analytics', '~> 3.0'
```

This is my package.json:

    "appcenter": "3.1.2",
    "appcenter-analytics": "3.1.2",
    "appcenter-crashes": "3.1.2",

This _should_ be the correct dependency version pulled from Pods for a 3.1.2 (^3.0.0?) package:

```
  - AppCenter/Analytics (3.3.4):
    - AppCenter/Core
  - AppCenter/Core (3.3.4)
  - AppCenter/Crashes (3.3.4):
    - AppCenter/Core
  - AppCenterReactNativeShared (3.1.2):
    - AppCenter/Core (= 3.3.4)
```

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Delete project Pods folder and Podfile.lock
2. Ensure your package.json is pointing to an older version, e.g. 3.1.2
3. npm install (and pod install in the ios folder)

### **Details**

1. Which version of the App Center SDK are you using?
    - 3.1.2
2. Which OS version did you experience the issue on?
    - Don't matter
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Don't matter
4. What third party libraries are you using?
    - Don't matter
5. Run the following command and paste the output below: `react-native info`

System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 25.45 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.18.3 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.5, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 11.5/11E608c - /usr/bin/xcodebuild
  npmPackages:
    react: ^16.13.1 => 16.13.1 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    react-native-cli: 2.0.1

6. If you're developing for React Native iOS, run the following command and paste the output below: `pod --version`

1.9.3

7. Please enable verbose logging for your app using `[MSAppCenter setLogLevel: MSLogLevelVerbose];` before your call to `[AppCenterReactNative register];` (or any other SDK registration). For Android, use `AppCenter.setLogLevel(android.util.Log.VERBOSE);` before your `SoLoader.init` call (you will need to `import com.microsoft.appcenter.AppCenter;`). Include the logs here:
- N/A; iOS compile issue.

**Comment by CyberMew** [comment]
If people are encountering the same issue as me, 4 ways to solve it, pick your poison:

1. Commit a Podfile.lock with the correct versions
2. Modify the node_modules folders (if local build)
3. Upgrade to v4 if you can
4. Add the following to your Podfile to enforce the versions ourselves first:
```
  pod 'AppCenter/Analytics', '~> 3.0'
  pod 'AppCenter/Crashes', '~> 3.0'
```

**Comment by russelarms** [comment]
Hi @CyberMew, thanks for reporting this issue! We're looking into it.

**Comment by CyberMew** [comment]
Thanks @russelarms for the quick fixes, when will this be released?

**Comment by russelarms** [comment]
@CyberMew the patch is released under the 3.1.3 tag. Please check it out and let us know if you have any problems.
